libfreerdp/core/transport.c

```diff
@@ -160,8 +160,13 @@ static BOOL transport_default_attach(rdpTransport* transport, int sockfd)
 {
 	BIO* socketBio = NULL;
 	BIO* bufferedBio;
+	const rdpSettings* settings;
+	rdpContext* context = transport_get_context(transport);
 
-	WINPR_ASSERT(transport);
+	settings = context->settings;
+	WINPR_ASSERT(settings);
+
+	freerdp_tcp_set_keep_alive_mode(settings, sockfd);
```

Comment:
function returns a value, better check it (yes, currently always `TRUE` but that might change)